examples/thread_duel/main.c

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (C) 2020 TUBA Freiberg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <thread.h>
+#include <sched.h>
+
+#include <math.h>
+
+#if MODULE_ZTIMER_USEC
+    #include "ztimer/xtimer_compat.h"
+#else
+    #include "xtimer.h"
+#endif
+
+#include "sched_round_robin.h"
+
+#define PRINT_STEPS 10
+#define WORK_SCALE  1000
+
+static void bad_wait(uint32_t us)
+{
+    uint32_t start = xtimer_now_usec();
+    /*keep the CPU busy waiting for some time to pass simulate working*/
+    while ((xtimer_now_usec() - start) < us){
+        /*do nothing*/
+    }
+}
+
+static void nice_wait(uint32_t us)
+{
+    /*be nice give the CPU some time to do other things or rest*/
+    xtimer_usleep(us);
+}
+
+/*
+ * the following are threads that count and wait with different strategies and
+ * print their current count in steps.
+ * the ration of active (doing hard work like checking the timer)
+ * to passive (wait to be informed when a certain time is there) waiting
+ * is determined by there value given to the thread.
+ * the restless threads do never pause.
+ */
+
+void * thread_nicebreaks(void * d)
+{
+    xtimer_usleep(200 * 1000);  /*always be nice at start*/
+#ifdef DEVELHELP
+    const char *name = thread_get_active()->name;
+#else
+    int16_t pid = thread_getpid();
+#endif
+
+    uint32_t w = 0;
+    uint32_t work = *((uint32_t *) d);
+    if (work > 10) {
+        work = 5;
+    }
+    uint32_t rest = (10 - work);
+    uint32_t step = 0;
```

Comment:
This logic is not evident for me just from looking at it, can you give more details (comments)?